### PR TITLE
Pseudo-PHP Code for Feature Idea

### PR DESCRIPTION
**Pull request by CDainMiller**
Hey SB team. 

I spoke with JD about a feature idea and he wanted me to submit up some code for it. I don't have time to debug this right now so beware! It is not executing at all.

Idea:
- The account dropdown on the local theme dev tools should display accounts in a nested fashion (parent->[child, child, child], parent_two->[child, child])

Why?
- Doing this eases platform developers burden when they deploy theme files to the theme's associated accounts. Such a nested relationship would allow us to see in a much more obvious way what context we are in when submitting these themes up to production (speaking from the POV of a StageBloc platform developer of course).

To implement such a thing I just added some opt groups in the PHP file here. But I don't like having to do another query and another loop to get and display this data, maybe you guys can think of something better. Maybe some eager loading on the API could fix that. Or allowing for passing a ```?expand=children``` flag to accounts, just like you allow on ```/account/:id/playlists?expand=audio```., and if it is expanded you append the children array on there. Something like that would be cool. That way this data would already be here in this loop if we accessed up top with the flag. Just an idea.

### CODE REVIEW

**Comment by CDainMiller** [comment]
Correction: *Or allowing for passing an expand flag to accounts just like you do **audio playlists**.

**Comment by CDainMiller** [comment]
Hey guys, how do we feel about this addition? If you guys think this is a good idea, It'd be amazing if we could squeeze this very small change in by the weekend because we are going to be submitting our network themes hundreds of times for the re-launch of Earwolf.

File: index.php [hunk 1]
```diff
@@ -115,13 +115,30 @@
...
+        	// If an account has children display them as a sub-optgroup to the parent, and if not continue on...
+        	$children = $stagebloc->post("/account/'. $account->id .'/children", array());
+        	if ( $children ) {
+        		$accountOptionsHTML .= '<optgroup label="' . $account->name . '">';
+        		foreach ( $children as $childAccount ) {
+        			$accountOptionsHTML .= '<option value="' . $childAccount->id . '" ' . ( $accountToUse == $childAccount->id ? 'selected' : '' ) . '>' . $childAccount->ins . '</option>';        			
+        		}
+        		$accountOptionsHTML .= '</optgroup>';
+        	else {
+            	$accountOptionsHTML .= '<option value="' . $account->id . '" ' . ( $accountToUse == $account->id ? 'selected' : '' ) . '>' . $account->ins . '</option>';
```
Comment (joshholat) [inline index.php]: This tabbing seems messed up.

File: index.php [hunk 2]
```diff
@@ -115,13 +115,30 @@
...
 
         $accountToUse = ( isset($_COOKIE['account']) ? $_COOKIE['account'] : $accounts[0]->id ); // Default to use the first account
-        foreach ( $accounts as $account ) // Build a <select> drop down to use to switch between accounts quickly
+
+        // Build a <select> drop down to use to switch between accounts quickly
+        foreach ( $accounts as $account ) 
         {
-            $accountOptionsHTML .= '<option value="' . $account->id . '" ' . ( $accountToUse == $account->id ? 'selected' : '' ) . '>' . $account->name . '</option>';
+        	// If an account has children display them as a sub-optgroup to the parent, and if not continue on...
+        	$children = $stagebloc->post("/account/'. $account->id .'/children", array());
```
Comment (joshholat) [inline index.php]: Does this work having double quotes to start and single quotes to end?

File: index.php (hunk 1 above)
Comment (CDainMiller) [inline index.php]: Yeah I wrote in my comment on this PR that this PHP isnt correct. This is just for the idea. JD told me to push it up. One of your guys has to review it though. But yeah the tabbing here is messed up.

**Comment by CDainMiller** [comment]
Do you think this is still something we could do @joshholat ? I still think it'd be nice to order the dropdown in nested fashion based on parent/child account relationships. :)

**Pull request closed without merging**